drivers/spi/spi_ll_stm32.c

```diff
@@ -51,6 +51,33 @@ LOG_MODULE_REGISTER(spi_ll_stm32);
 #endif /* CONFIG_SOC_SERIES_STM32MP1X */
 
 #ifdef CONFIG_SPI_STM32_DMA
+
+#ifdef CONFIG_SOC_SERIES_STM32H7X
+
+#define IS_NOCACHE_MEM_REGION(node_id)                                          \
+	COND_CODE_1(DT_ENUM_HAS_VALUE(node_id, zephyr_memory_attr, RAM_NOCACHE),    \
+			(true),                                                             \
+			(false))
+
+#define GET_MEM_REGION(node_id)                                                 \
+	{                                                                           \
+		.start = (const uintptr_t)DT_REG_ADDR(node_id),                         \
+		.end =                                                                  \
+			(const uintptr_t)(DT_REG_ADDR(node_id) + DT_REG_SIZE(node_id)) - 1, \
+		.isnocache = IS_NOCACHE_MEM_REGION(node_id)                             \
```

Comment:
if you are interested only in the `NOCACHE` regions, why not filtering out the other regions when constructing `mem_regions[]`?

Reply:
I think you mean using `COND_CODE_1` in combination with `IS_NOCACHE_MEM_REGION` to not add "no nocache" entries to the array? That would be ideal, I will try.

Reply:
Done